machine_test.go

```diff
@@ -814,6 +834,45 @@ func TestCaptureFifoToFile(t *testing.T) {
 	}
 }
 
+func TestCaptureFifoToFile_nonblock(t *testing.T) {
+	fifoPath := filepath.Join(testDataPath, "fifo")
+
+	if err := syscall.Mkfifo(fifoPath, 0700); err != nil {
+		t.Fatalf("Unexpected error during syscall.Mkfifo call: %v", err)
+	}
+	defer os.Remove(fifoPath)
+
+	logPath := fifoPath + ".log"
+	logFile, err := os.OpenFile(logPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		t.Fatalf("Failed to create log file: %v", err)
+	}
+
+	if err := captureFifoToFile(fctesting.NewLogEntry(t), fifoPath, logFile); err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	defer os.Remove(logPath)
+
+	f, err := os.OpenFile(fifoPath, os.O_RDWR, 0600)
+	if err != nil {
+		t.Fatalf("Failed to open file, %q: %v", fifoPath, err)
+	}
+
+	f.Write([]byte("Hello world!"))
+	defer f.Close()
+
+	go func() {
+		t := time.NewTicker(250 * time.Millisecond)
+		select {
+		case <-t.C:
+			f.Close()
+		}
+	}()
+
+	assert.NoError(t, err, "failed to stat file")
```

Comment:
You're missing the actual stat here I think.  You could also use `ioutil.ReadFile` to get the file content and compare it to what you expect ("Hello world!").

Reply:
doh, fixed.

Reply:
Also, to test the expected bytes, we'd need to write as soon as `captureFifoToFile` opens the file for reading, otherwise the bytes get lost.

Reply:
Alright, added the assert for expected bytes.